domain/containerimageresourcestore/service/service.go

```diff
@@ -121,3 +127,31 @@ func (s Service) Remove(
 	}
 	return nil
 }
+
+// getSizeAndFingerprint gets the size and fingerprint that a DockerImageDetails
+// struct will have when exported as a blob from the container image metadata
+// service.
+func getSizeAndFingerprint(dockerDetails docker.DockerImageDetails) (int64, store.Fingerprint, error) {
+	data, err := json.Marshal(dockerDetails)
+	if err != nil {
+		return 0, store.Fingerprint{}, errors.Errorf("marshelling container image metadata: %w", err)
+	}
+	reader := bytes.NewReader(data)
+	fingerprint, err := charmresource.GenerateFingerprint(reader)
+	if err != nil {
+		return 0, store.Fingerprint{}, errors.Errorf("generating fingerprint: %w", err)
+	}
+	return int64(reader.Len()), store.NewFingerprint(fingerprint.Fingerprint), nil
```

Comment:
`Len returns the number of bytes of the unread portion of the slice.`

Reply:
Ah, I thought I had fixed this, I only caught it in QA because I'd made the same mistake in the tests.